File: 121-best-time-to-buy-and-sell-stock/121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        l = 0
        res = 0
        for r in range(1, len(prices)):
            res = max(res, prices[r] - prices[l])
            if prices[r] < prices[l]:
                l = r
            
        return res

File: 121-best-time-to-buy-and-sell-stock/121-best-time-to-buy-and-sell-stock.c


int maxProfit(int* prices, int pricesSize){
    int l = 0;
    int res = 0;
    
    for (int r = 1; r < pricesSize; r++) {
        if (prices[r] - prices[l] > res) {
            res = prices[r] - prices[l];
        }
        if (prices[r] < prices[l]) {
            l = r;
        }
    }
    
    return res;
}